File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int cnt=1;
        int a=nums[0];
        for(int i=1;i<nums.length;i++){

            if(a==nums[i])cnt++;
            else cnt--;
            if(cnt==0){
                a=nums[i];
                cnt=1;
            }
        }
        return a;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        ele=nums[0]
        count=1
        for i in nums:
            if i == ele : count=count+1
            else : count=count-1

            if count==0: 
                ele=i
                count=1

        
        return ele
            